File: backend/demo/src/main/java/com/traffic/demo/entity/Fine.java
package com.traffic.demo.entity;

import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "fine")
public class Fine {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String referenceNumber;

    @Column(nullable = false)
    private Double amount;

    @Column(nullable = false)
    private String status;

    @Column(nullable = false)
    private Long officerId;

    @Column(nullable = false)
    private Long driverId;

    @Column(nullable = false)
    private LocalDate fineDate;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private FineCategory category;

    public Long getId() {
        return id;
    }

    public String getReferenceNumber() {
        return referenceNumber;
    }

    public Double getAmount() {
        return amount;
    }

    public String getStatus() {
        return status;
    }

    public Long getOfficerId() {
        return officerId;
    }

    public Long getDriverId() {
        return driverId;
    }

    public LocalDate getFineDate() {
        return fineDate;
    }

    public FineCategory getCategory() {
        return category;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setReferenceNumber(String referenceNumber) {
        this.referenceNumber = referenceNumber;
    }

    public void setAmount(Double amount) {
        this.amount = amount;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void setOfficerId(Long officerId) {
        this.officerId = officerId;
    }

    public void setDriverId(Long driverId) {
        this.driverId = driverId;
    }

    public void setFineDate(LocalDate fineDate) {
        this.fineDate = fineDate;
    }

    public void setCategory(FineCategory category) {
        this.category = category;
    }
}

File: mobile-app/lib/models/fine.dart
class Fine {
  final int id;
  final String referenceNumber;
  final String categoryName;
  final double amount;
  final String status;
  final int officerId;
  final int driverId;
  final String? fineDate;

  Fine({
    required this.id,
    required this.referenceNumber,
    required this.categoryName,
    required this.amount,
    required this.status,
    required this.officerId,
    required this.driverId,
    this.fineDate,
  });

  factory Fine.fromJson(Map<String, dynamic> json) {
    return Fine(
      id: json['id'] ?? 0,
      referenceNumber: json["referenceNumber"] ?? "",
      categoryName: json["categoryName"] ?? "",
      amount: (json["amount"] ?? 0).toDouble(),
      status: json["status"] ?? "NOT_PAID",
      officerId: json["officerId"] ?? 0,
      driverId: json["driverId"] ?? 0,
      fineDate: json["fineDate"]?.toString(),
    );
  }
}